src/main/java/ch/njol/skript/util/SoundUtils.java

```diff
@@ -1,49 +1,65 @@
-/**
- *   This file is part of Skript.
- *
- *  Skript is free software: you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  Skript is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
- *
- * Copyright Peter Güttinger, SkriptLang team and contributors
- */
 package ch.njol.skript.util;
 
+import ch.njol.skript.Skript;
+import org.bukkit.NamespacedKey;
 import org.bukkit.Sound;
 import org.jetbrains.annotations.Nullable;
 
-/**
- * @author Peter Güttinger
- */
-public abstract class SoundUtils {
-	private SoundUtils() {}
-	
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Locale;
+
+public class SoundUtils {
+
+	private static final boolean IS_INTERFACE;
+	private static final Method VALUE_OF_METHOD;
+	private static final Method GET_KEY_METHOD;
+
 	static {
-		assert false;
+		try {
+			Class<?> SOUND_CLASS = Class.forName("org.bukkit.Sound");
+			IS_INTERFACE = SOUND_CLASS.isInterface();
+			VALUE_OF_METHOD = SOUND_CLASS.getDeclaredMethod("valueOf", String.class);
+			if (Skript.methodExists(SOUND_CLASS, "getKey")) {
+				// I believe this method was added around Bukkit 1.16
+				// This is only added to make tests not fail when testing MC 1.15 and below
+				GET_KEY_METHOD = SOUND_CLASS.getDeclaredMethod("getKey");
+			} else {
+				GET_KEY_METHOD = null;
+			}
+		} catch (NoSuchMethodException | ClassNotFoundException e) {
+			throw new RuntimeException(e);
+		}
 	}
-	
-	private final static EnumUtils<Sound> util = new EnumUtils<>(Sound.class, "sounds");
-	
+
+	/**
+	 * Get the NamespacedKey of a Bukkit Sound enum
+	 *
+	 * @param soundString String to check for enum
+	 * @return Sound key if available else null
+	 */
+	@SuppressWarnings("deprecation")
 	@Nullable
-	public static Sound parse(final String s) {
-		return util.parse(s);
+	public static NamespacedKey getSoundKeyFromEnum(String soundString) {
```

Comment:
this is only used in EffPlaySound so you might as well just move it into there

Reply:
I left it in the SoundUtil class as I felt like it would be cleaner.
This keeps all the ugly reflection stuff in one spot.
When Skript finally is 1.21.x+, it will be easier to strip out this useless stuff.